src/daemon/control.rs

```diff
@@ -263,19 +263,59 @@ fn presigned_txs_list_from_outpoints(
         db_vaults_min_status(db_path, VaultStatus::Funded)?
     };
 
+    // For each presigned transaction, append it as well as its extracted version if it's final.
     let mut tx_list = Vec::with_capacity(db_vaults.len());
     for db_vault in db_vaults {
         let outpoint = db_vault.deposit_outpoint;
 
-        let (_, unvault) = db_unvault_transaction(db_path, db_vault.id)?;
+        let (_, unvault_psbt) = db_unvault_transaction(db_path, db_vault.id)?;
+        let mut finalized_unvault = unvault_psbt.clone();
+        let unvault = VaultPresignedTransaction {
+            transaction: if finalized_unvault.finalize(&revaultd.secp_ctx).is_ok() {
```

Comment:
nit: I haven't tested it but I think you can just use map
```
transaction: finalized_unvault.finalize(&revaultd.secp_ctx).map(|f| f.into_psbt().extract_tx())
```

Reply:
Nope because finalize modifies the tx in place and does not return it